XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -482,23 +482,31 @@ public Collider Collider
         {
             get
             {
+                var isSet = false;
+
                 if (BoundingBox != null &&
                     BoundingBox.BoundingBoxCollider != null)
                 {
-                    return BoundingBox.BoundingBoxCollider;
+                    thisCollider = BoundingBox.BoundingBoxCollider;
+                    isSet = true;
                 }
 
                 if (thisCollider == null)
                 {
                     thisCollider = gameObject.GetComponent<Collider>();
+                    isSet = true;
                 }
 
                 if (thisCollider == null)
                 {
                     thisCollider = gameObject.EnsureComponent<BoxCollider>();
+                    isSet = true;
                 }
 
-                transform.GetColliderBounds();
+                if (isSet)
+                {
+                    transform.GetColliderBounds(ref cachedColliders);
```

Comment:
We should probably call this on enable as well

Reply:
Sure, but also - do we need to run this in the Getter? I'm feeling weird about putting potentially (relatively) long-running code there. People don't assume getters do a lot

Reply:
Let me rephrase: could we move this out?

Reply:
Yeah I'm not liking how this property is setup atm. But the purpose of all the logic is to ensure that the object is properly setup, this ensures we always will get back out a valid value, if the dependencies aren't setup. It's always understood that a Property can be doing additional work. That's part of what makes encapsulation so powerful, and why many people frown upon directly accessing public fields unless they're readonly or constant.